lib/stimpack/integrations/rspec.rb

```diff
@@ -1,19 +1,46 @@
 module Stimpack
   module Integrations
     class RSpec
-      def self.install(app)
-        return unless defined?(::RSpec)
-        # Sometimes, the `rspec-rails` gem is installed in the development
-        # group. This means the ::RSpec module will be defined. However, this
-        # doesn't mean that we've loaded rspec-core, or even about to run tests.
-        # Let's make sure we are actually loading the test environment before
-        # installing our integration. An easy of doing this is seeing of the
-        # configuration method exists on RSpec.
-        return unless ::RSpec.respond_to?(:configuration)
+      def initialize
+        # This is the list of directories RSpec was told to run.
+        to_run = ::RSpec.configuration.instance_variable_get(:@files_or_directories_to_run)
+        default_path = ::RSpec.configuration.default_path
 
-        Packs.each do |pack|
-          ::RSpec.configuration.pattern.concat(",#{pack.relative_path.join("spec/**/*_spec.rb")}")
+        if to_run == [default_path]
+          # This is the default case when you run `rspec`. We want to add all the pack's spec paths
+          # to the collection of directories to run.
+
+          pack_paths = Packs.all.map do |pack|
+            spec_path = pack.relative_path.join(default_path)
+            spec_path.to_s if spec_path.exist?
+          end
+
+          to_run.concat(pack_paths)
+        else
+          # This is when `rspec` is run with a list of directories or files. We scan this list to see
+          # if any of them matches a pack's directory. If it does, we concat the `default_path` to the
+          # end of it.
+          #
+          # packs/my_pack => packs/my_pack/spec
+          # 
+          # If it doesn't match a pack path, we leave it alone.
+
+          to_run.map! do |path|
+            if pack = Packs.all_by_path[path]
+              [
+                pack,
+                *pack.all_children
```

Comment:
This is great -- I do think documenting this in the README would be helpful!

Reply:
I'm not entirely sure which methods we want to make public/private yet.

Reply:
This seems key though in terms of the behavior of rspec based on what args are passed in.

Reply:
I'm specifically referring to:

bundle exec rspec #=> Runs _all_ specs in your entire application (packs and all)
bundle exec rspec spec/some/specific_spec.rb #=> Runs just that one test
bundle exec rspec packs/foobar #=> Runs all specs under the foobar pack _and_ all the specs of its nested packs
bundle exec rspec packs/foobar pack/binbaz #=> same as above but also adds the binbaz pack
bundle exec rspec packs/foobar/spec #=> Runs all spec files inside the "packs/foobar/spec" directory
bundle exec rspec packs/foobar/nested_pack #=> Runs all specs under the "packs/foobar/nested_pack" pack

Reply:
Done!